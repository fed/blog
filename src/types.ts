import { MarkdownRemark } from '../graphql-types';

export enum Category {
    BACONJS = 'baconjs',
    RXJS = 'rxjs',
    JAVASCRIPT = 'javascript',
    REACT = 'react',
    TESTING = 'testing',
    ACCESSIBILITY = 'accessibility',
    BROWSERS = 'browsers',
    GENERAL = 'general',
}

// export type Post = MarkdownRemark;

export interface Post {
    id: string;
    html: string;
    timeToRead: number;
    frontmatter: {
        title: string;
        date: string;
        spoiler: string;
        category: Category;
    };
    fields: {
        slug: string;
    };
}
