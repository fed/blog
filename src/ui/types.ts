export enum CategoryId {
    GENERAL = 'general',
    DOM = 'dom',
    JAVASCRIPT = 'javascript',
    TESTING = 'testing',
    ACCESSIBILITY = 'accessibility',
    DECLARATIVE = 'declarative',
    FRP = 'frp',
}

export interface Post {
    id: string;
    title: string;
    spoiler: string;
    date: string;
    categoryId: CategoryId;
    slug: string;
}
