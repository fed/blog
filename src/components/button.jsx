import React from 'react';
import styled from 'styled-components';

import { baseFocusStateStyles, baseSansSerifStyles } from '../styles/mixins';

const BaseButton = styled.a`
    ${baseFocusStateStyles};
    ${baseSansSerifStyles};
    border: 0;
    border-radius: 6px;
    color: #fff;
    cursor: pointer;
    display: inline-block;
    font-size: 14px;
    font-weight: 400;
    padding: 10px 15px 8px;
    position: relative;
    text-decoration: none;
    text-transform: uppercase;
    transition: all 0.2s ease;
    :active {
        transform: translate(0, 3px);
    }
    :hover {
        border: inherit;
    }
    :focus {
        border: 1px solid #fff;
        padding: 10px 15px;
    }
`;

const TwitterButton = styled(BaseButton)`
    background-color: #2ca9e1;
    box-shadow: 0 6px 0 #2695bc;
    :active {
        box-shadow: 0 2px 0 #2695bc;
    }
`;

const GitHubButton = styled(BaseButton)`
    background-color: #353631;
    box-shadow: 0 6px 0 #141412;
    :active {
        box-shadow: 0 2px 0 #141412;
    }
`;

const MediumButton = styled(BaseButton)`
    background-color: #4caf50;
    box-shadow: 0 6px 0 #388e3c;
    :active {
        box-shadow: 0 2px 0 #388e3c;
    }
`;

const CodePenButton = styled(BaseButton)`
    background-color: #657e91;
    box-shadow: 0 6px 0 #374957;
    :active {
        box-shadow: 0 2px 0 #374957;
    }
`;

const LinkedInButton = styled(BaseButton)`
    background-color: #0073b2;
    box-shadow: 0 6px 0 #005377;
    :active {
        box-shadow: 0 2px 0 #005377;
    }
`;

export default function Button(props) {
    let BrandedButton;

    switch (props.type) {
        case 'twitter':
            BrandedButton = TwitterButton;
            break;
        case 'github':
            BrandedButton = GitHubButton;
            break;
        case 'medium':
            BrandedButton = MediumButton;
            break;
        case 'codepen':
            BrandedButton = CodePenButton;
            break;
        case 'linkedin':
            BrandedButton = LinkedInButton;
            break;
    }

    if (!BrandedButton) {
        return null;
    }

    return (
        <BrandedButton
            className={props.className}
            href={props.url}
            target="_blank"
            rel="noopener noreferrer"
        >
            {props.children}
        </BrandedButton>
    );
}
