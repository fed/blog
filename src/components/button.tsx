import React from 'react';
import styled from 'styled-components';

import { baseFocusStateStyles, fontFamilySansSerif } from '../styles/mixins';
import { Social } from '../types';

const BaseButton = styled.a`
    ${baseFocusStateStyles};
    border: 0;
    border-radius: 6px;
    color: #fff;
    cursor: pointer;
    display: inline-block;
    font-family: ${fontFamilySansSerif};
    font-size: 14px;
    font-weight: 400;
    padding: 10px 15px 8px;
    position: relative;
    text-decoration: none;
    text-transform: uppercase;
    transition: all 0.2s ease;
    :active {
        transform: translate(0, 3px);
    }
    :hover {
        border: inherit;
    }
    :focus {
        border: 1px solid #fff;
        padding: 10px 15px;
    }
`;

const TwitterButton = styled(BaseButton)`
    background-color: #2ca9e1;
    box-shadow: 0 6px 0 #2695bc;
    :active {
        box-shadow: 0 2px 0 #2695bc;
    }
`;

const GitHubButton = styled(BaseButton)`
    background-color: #353631;
    box-shadow: 0 6px 0 #141412;
    :active {
        box-shadow: 0 2px 0 #141412;
    }
`;

const MediumButton = styled(BaseButton)`
    background-color: #4caf50;
    box-shadow: 0 6px 0 #388e3c;
    :active {
        box-shadow: 0 2px 0 #388e3c;
    }
`;

const CodePenButton = styled(BaseButton)`
    background-color: #657e91;
    box-shadow: 0 6px 0 #374957;
    :active {
        box-shadow: 0 2px 0 #374957;
    }
`;

const LinkedInButton = styled(BaseButton)`
    background-color: #0073b2;
    box-shadow: 0 6px 0 #005377;
    :active {
        box-shadow: 0 2px 0 #005377;
    }
`;

interface Props {
    url?: string;
    type?: Social;
    children?: React.ReactNode;
    className?: string;
}

export const Button: React.FC<Props> = ({ type, url, children, className }) => {
    let BrandedButton;

    switch (type) {
        case Social.TWITTER:
            BrandedButton = TwitterButton;
            break;
        case Social.GITHUB:
            BrandedButton = GitHubButton;
            break;
        case Social.MEDIUM:
            BrandedButton = MediumButton;
            break;
        case Social.CODEPEN:
            BrandedButton = CodePenButton;
            break;
        case Social.LINKEDIN:
            BrandedButton = LinkedInButton;
            break;
    }

    if (!BrandedButton) {
        return null;
    }

    return (
        <BrandedButton className={className} href={url} target="_blank" rel="noopener noreferrer">
            {children}
        </BrandedButton>
    );
};
