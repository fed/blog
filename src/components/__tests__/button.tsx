import React from 'react';
import renderer from 'react-test-renderer';

import { Social } from '../../types';
import { Button } from '../button';

describe('Button', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Button type={Social.TWITTER} url="https://twitter.com/fknussel">
                    Twitter
                </Button>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
