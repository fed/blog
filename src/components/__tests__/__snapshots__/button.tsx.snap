// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<a
  className="sc-bdvvaa sc-gsDJrp bhLkGr gycrnD"
  href="https://twitter.com/fknussel"
  rel="noopener noreferrer"
  target="_blank"
>
  Twitter
</a>
`;
